### PR TITLE
feat(CLI): Add '--detail' for detail version information (#283)

### LINKED ISSUE #283: feat(CLI): envd doctor command
## Description

envd will print out status information in `envd doctor`:

- envd version
- Operating system
- Docker host and version
- Kubernetes version
- Type of Kubernetes cluster (Docker for Mac, Microk8s, etc)
- Container runtime

### PR DESCRIPTION
**Pull request by popfido**
## What does this PR do?

- Add extra doctor information in `version` by `--detail`

fix #283 

## Does your PR introduce any breaking changes? If yes, please list them.

No.

## Before submitting

- [x] Was this **discussed/approved** via a issue? (not for typos and docs)
- [x] Did you make sure your **PR does only one thing**, instead of bundling different changes together?
- [ ] Did your make sure to **update the documentation** with your changes? (if necessary)
- [ ] Did you write any **new necessary tests**? (not for typos and docs)
- [x] Did you verify new and **existing tests pass** locally with your changes?
- [x] Did you list all the **breaking changes** introduced by this pull request?

Signed-off-by: Popfido <wanghailin317@gmail.com>

### CODE REVIEW

**Review by gaocegege — APPROVED** [review]
LGTM!

Thanks for your contribution! :tada: :+1:

**Review by gaocegege — APPROVED** [review]

**Comment by popfido** [comment]
cc: @gaocegege . Don't know whether the failure in CI/lint need to be fixed.

File: pkg/version/version.go [hunk 1]
```diff
@@ -53,12 +59,26 @@ type Version struct {
...
+	OSVersion         string
+	OSType            string
+	KernelVersion     string
+	Architecture      string
+	DockerVersion     string
+	ContainerRuntimes string
+	DefaultRuntime    string
+}
+
+type generalEngine struct {
```
Comment (gaocegege) [inline pkg/version/version.go:72]: The struct is unused, I think we can remove it.

**Review by gaocegege — COMMENTED** [review]
After the update, please rerun `go mod tidy` again.

The command `go mod tidy` will update the dependencies (In our case, it will remove `github.com/pkg/errors` in go.mod)

**Review by gaocegege — APPROVED** [review]

**Comment by gaocegege** [comment]
Thanks for your contribution! :tada: :+1:

**Comment by gaocegege** [comment]
Please add @popfido for the code contribution. @all-contributors

**Comment by gaocegege** [comment]
@all-contributors Please add @popfido for the code contribution

**Pull request merged**